hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/onset/WorkflowOnsetTest.java

```diff
@@ -0,0 +1,298 @@
+class WorkflowOnsetTest {
+
+    @Mock private OnsetChecker checker;
+
+    @Mock(strictness = LENIENT)
+    private Parser<Query> queryParser;
+
+    @Mock(strictness = LENIENT)
+    private Parser<Transaction> txParser;
+
+    @Mock(strictness = LENIENT)
+    private Parser<SignedTransaction> signedParser;
+
+    @Mock(strictness = LENIENT)
+    private Parser<TransactionBody> txBodyParser;
+
+    private SessionContext ctx;
+
+    private Transaction tx;
+    private ByteString signedTransactionBytes;
+    private SignatureMap signatureMap;
+    private SignedTransaction signedTx;
+    private ByteString bodyBytes;
+    private TransactionBody txBody;
+
+    private ByteBuffer inputBuffer;
+    private WorkflowOnset onset;
+
+    @BeforeEach
+    void setup() throws InvalidProtocolBufferException {
+        inputBuffer = ByteBuffer.allocate(0);
+
+        final var content = ConsensusCreateTopicTransactionBody.newBuilder().build();
+        txBody = TransactionBody.newBuilder().setConsensusCreateTopic(content).build();
+
+        bodyBytes = ByteString.copyFrom("bodyBytes", StandardCharsets.UTF_8);
+        signatureMap = SignatureMap.newBuilder().build();
+        signedTx =
+                SignedTransaction.newBuilder()
+                        .setBodyBytes(bodyBytes)
+                        .setSigMap(signatureMap)
+                        .build();
+
+        signedTransactionBytes =
+                ByteString.copyFrom("signedTransactionBytes", StandardCharsets.UTF_8);
+
+        tx = Transaction.newBuilder().setSignedTransactionBytes(signedTransactionBytes).build();
+
+        when(txBodyParser.parseFrom(bodyBytes)).thenReturn(txBody);
+        when(signedParser.parseFrom(signedTransactionBytes)).thenReturn(signedTx);
+        when(txParser.parseFrom(inputBuffer)).thenReturn(tx);
+
+        ctx = new SessionContext(queryParser, txParser, signedParser, txBodyParser);
+        onset = new WorkflowOnset(checker);
+    }
+
+    @SuppressWarnings("ConstantConditions")
+    @Test
+    void testConstructorWithIllegalArguments() {
+        assertThatThrownBy(() -> new WorkflowOnset(null)).isInstanceOf(NullPointerException.class);
+    }
+
+    @Test
+    void testParseAndCheckSucceeds() throws PreCheckException {
+        // when
+        final var result = onset.parseAndCheck(ctx, inputBuffer);
+
+        // then
+        assertThat(result.txBody()).isEqualTo(txBody);
+        assertThat(result.signatureMap()).isEqualTo(signatureMap);
+        assertThat(result.functionality()).isEqualTo(HederaFunctionality.ConsensusCreateTopic);
+    }
+
+    @Test
+    void testParseAndCheckWithByteArraySucceeds(@Mock Parser<Transaction> localTxParser)
+            throws PreCheckException, InvalidProtocolBufferException {
+        // when
+        final var byteArray = new byte[0];
+        when(localTxParser.parseFrom(byteArray)).thenReturn(tx);
+
+        ctx = new SessionContext(queryParser, localTxParser, signedParser, txBodyParser);
```

Comment:
Is there an advantage to having `ctx` as a class variable here?

Reply:
No, fixed it.